includes/api/class-edd-api-v2.php

```diff
@@ -148,4 +148,152 @@ public function get_products( $args = array() ) {
 		return $products;
 	}
 
+	/**
+	 * Process Get Customers API Request
+	 *
+	 * @access public
+	 * @since 2.6
+	 * @global object $wpdb Used to query the database using the WordPress Database API
+	 * @param array $args Array of arguments for filters customers
+	 * @return array $customers Multidimensional array of the customers
+	 */
+	public function get_customers( $args = array() ) {
+		global $wpdb;
+
+		$paged    = $this->get_paged();
+		$per_page = $this->per_page();
+		$offset   = $per_page * ( $paged - 1 );
+
+		$defaults = array(
+			'customer'  => null,
+			'date'      => null,
+			'startdate' => null,
+			'enddate'   => null,
+			'number'    => $per_page,
+			'offset'    => $offset,
+		);
+
+		$args      = wp_parse_args( $args, $defaults );
+		$customers = array();
+		$error     = array();
+
+		if( ! user_can( $this->user_id, 'view_shop_sensitive_data' ) && ! $this->override ) {
+			return $customers;
+		}
+
+		if( is_numeric( $args['customer'] ) ) {
+			$field = 'id';
+		} else {
+			$field = 'email';
+		}
+
+		$args[ $field ] = $args['customer'];
+
+		$dates = $this->get_dates( $args );
+
+		if( $args['date'] === 'range' ) {
+
+			// Ensure the end date is later than the start date
+			if( ( ! empty( $args['enddate'] ) && ! empty( $args['enddate'] ) ) && $args['enddate'] < $args['startdate'] ) {
+				$error['error'] = __( 'The end date must be later than the start date!', 'easy-digital-downloads' );
+			}
+
+			$date_range = array();
+			if ( ! empty( $args['startdate'] ) ) {
+				$date_range['start'] = $dates['year']     . sprintf('%02d', $dates['m_start'] ) . $dates['day_start'];
+			}
+
+			if ( ! empty( $args['enddate'] ) ) {
+				$date_range['end'] = $dates['year_end'] . sprintf('%02d', $dates['m_end'] )   . $dates['day_end'];
+			}
+
+			$args['date'] = $date_range;
+
+		} else {
+			if( $args['date'] == 'this_quarter' || $args['date'] == 'last_quarter'  ) {
```

Comment:
@pippinsplugins I don't see `this_week` or `last_week` currently supported in the `get_dates` method of the existing API.